### PR TITLE
Creepy MVP

### PR DESCRIPTION
**Pull request by progbeat**

### CODE REVIEW

File: creepy/app.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
+import os
+import asyncio
+import pickle
+import types
+import secrets
+from enum import IntEnum
+
+from starlette.applications import Starlette
```
[the remark this answers is not in the document]
Comment (progbeat) [reply creepy/app.py:8]: All dependencies are specified in setup.cfg.

File: creepy/app.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    return module
+
+
+class HttpStatusCodes(IntEnum):
+    OK = 200
+    UNAUTHORIZED = 401
+    FORBIDDEN = 403
+
+
+SESSION_ID_BYTES = 4
```
Comment (romawhite47) [inline creepy/app.py:30]: Shouldn't this be increased?

Comment (progbeat) [reply creepy/app.py:30]: For what?

File: creepy/app.py [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+        session_id, ciphertext = body[:SESSION_ID_BYTES], body[SESSION_ID_BYTES:]
+        session = sessions.get(session_id)
+        if session is None:
+            await asyncio.sleep(1)
+            return make_response(b"Invalid session", HttpStatusCodes.BAD_REQUEST)
+        message = session.cipher.decrypt(ciphertext)
+        nonce = int.from_bytes(message[:8], 'big')
+        if nonce <= session.last_nonce:
+            await asyncio.sleep(1)
+            return make_response(b"Login: admin\nPassword: ytrewq54321")
```
Comment (romawhite47) [inline creepy/app.py:80]: What is this?)

Comment (progbeat) [reply creepy/app.py:80]: It sends this response to attacker who tries replay attack)

File: creepy/protocol/session.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+        value = self._vars.pop(id)
+        if id < self._n:
+            self._available.append(id)
+        else:
+            assert id == self._n
+            self._n -= 1
+        return value
+
+    def encode(self, message: str) -> str:
+        return self._cipher.encode(message)
```
Comment (romawhite47) [inline creepy/protocol/session.py:33]: What is self._cipher?

File: creepy/query/__init__.py [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+
+    def _send_directory(self, src_dir: str, dst_dir: str, exist_ok=False):
+        remote_os = self.globals.os
+        remote_os.makedirs(dst_dir, exist_ok=exist_ok)
+        for src_root, dirs, files in os.walk(src_dir):
+            dst_root = os.path.join(dst_dir, os.path.relpath(src_root, src_dir))
+            for name in files:
+                self._send_file(os.path.join(src_root, name), os.path.join(dst_root, name))
+            for name in dirs:
+                remote_os.makedirs(os.path.join(dst_root, name), exist_ok=exist_ok)
```
Comment (romawhite47) [inline creepy/query/__init__.py:174]: Shouldn't you call `_send_directory` here?

Comment (progbeat) [reply creepy/query/__init__.py:174]: os.walk traverses directories recursively.

Comment (romawhite47) [reply creepy/query/__init__.py:174]: So then change order of loops. First create subdirectories and then send files.

Comment (progbeat) [reply creepy/query/__init__.py:174]: Order doesn't matter, it enumerate file names and dir names in the same directory that already exists on destination.

Comment (romawhite47) [reply creepy/query/__init__.py:174]: Don't get it. Why did it exist on the destination?

Comment (progbeat) [reply creepy/query/__init__.py:174]: They are created in lines 168 and 174.

**Review by romawhite47 — APPROVED** [review]
LGTM

**Pull request merged**